client/state/terms/schema.js

```diff
@@ -29,3 +29,23 @@ export const termsSchema = {
 	},
 	additionalProperties: false
 };
+
+export const queriesSchema = {
+	type: 'object',
+	patternProperties: {
+		'^\\d+$': {
+			type: 'object',
+			patternProperties: {
+				// Queries are JSON strings, prepended by a site ID and taxonomy slug
+				'^([A-Za-z0-9-_]+:)\\{[^\\}]*\\}$': {
```

Comment:
The more I work with "serialized keys", the more I want to limit their usage. It's a bit of a (necessary) hack, and the prefixing especially seems to be trying to recreate the structure of an object, when in most cases we can simply structure it as an object. So I'd suggest `{ siteId: { taxonomy: { serializedQuery: [ results ] } } }` . Added bonus is consistency with your `items` reducer.